plugins/modules/packaging/os/snap.py

```diff
@@ -186,36 +266,92 @@ def is_snap_enabled(self, snap_name):
+                commands = [params + [{'actionable_snaps': [s]}] for s in actionable_snaps]
+            else:
+                commands = [params + [{'actionable_snaps': actionable_snaps}]]
+            self.vars.cmd, rc, out, err = self._run_multiple_commands(commands)
+
+            if rc != 0:
+                classic_snap_pattern = re.compile(r'^error: This revision of snap "(?P<package_name>\w+)"'
+                                                  r' was published using classic confinement')
+                match = classic_snap_pattern.match(err)
+                if match:
+                    err_pkg = match.group('package_name')
+                    msg = "Couldn't install {name} because it requires classic confinement".format(name=err_pkg)
+                else:
+                    msg = "Ooops! Snap installation failed while executing '{cmd}', please examine logs and " \
+                          "error output for more details.".format(cmd=self.vars.cmd)
+                raise ModuleHelperException(msg=msg)
+
+        self.set_options()
+
+    def set_options(self):
+        actionable_snaps = [s for s in self.vars.name if self.is_snap_installed(s)]
+        overall_options_changed = []
+
+        if self.vars.options is not None:
+            for snap_name in actionable_snaps:
+                option_map = self.retrieve_option_map(snap_name=snap_name)
+
+                options_changed = []
+
+                for option_string in self.vars.options:
+                    match = self.__set_param_re.match(option_string)
+
+                    if not match:
+                        msg = "Cannot parse set option '{option_string}'".format(option_string=option_string)
+                        raise ModuleHelperException(msg)
+
+                    snap_prefix = match.group("snap_prefix")
+                    selected_snap_name = snap_prefix[:-1] if snap_prefix else None
+
+                    if selected_snap_name is not None and selected_snap_name not in self.vars.name:
+                        msg = "Snap option '{option_string}' refers to snap which is not in the list of snap names".format(option_string=option_string)
+                        raise ModuleHelperException(msg)
+
+                    if selected_snap_name is None or (snap_name is not None and snap_name == selected_snap_name):
+                        key = match.group("key")
+                        value = match.group("value")
+
+                        if key not in option_map or key in option_map and option_map[key] != value:
+                            option_without_prefix = key + "=" + value
+                            option_with_prefix = option_string if selected_snap_name is not None else snap_name + ":" + option_string
+                            options_changed.append(option_without_prefix)
+                            overall_options_changed.append(option_with_prefix)
+
+                if len(options_changed) > 0:
```

Comment:
More Pythonic would be:
```suggestion
                if options_changed:
```
(Same further below.)

Reply:
Although, personally, I don't like this "Pythonic" version at all, I checked that it is actually explicitly defined this way in the language definition.